### DISCUSSION TITLE
What should I do to get ECG_Q_Peaks and  ECG_S_Peaks?

### DISCUSSION THREAD

**Discussion by sdujicun** [neuropsychology/NeuroKit#458] [Q&A] [answered]
When I used the following two lines of code,
_, waves_peak = nk.ecg_delineate(ecg_signal, rpeaks, sampling_rate=3000)
plot = nk.events_plot([waves_peak['ECG_T_Peaks'],
                       waves_peak['ECG_P_Peaks'],
                       waves_peak['ECG_Q_Peaks'],
                       waves_peak['ECG_S_Peaks']], ecg_signal)

an error exist "KeyError: 'ECG_Q_Peaks'".

What should I do to get ECG_Q_Peaks and  ECG_S_Peaks?

**Comment by Tam-Pham** [comment, marked as answer]
Hi @sdujicun, 

Currently `nk.ecg_delineate()` has a derivative method and two other wavelet-based methods. The default method is `method="dwt"` (discrete wavelet method) and unfortunately, the returns of this method only include  P-peaks, T-peaks, P-onsets, P-offsets, T-onsets, T-offsets. For Q-peaks and S-peaks, you can use the derivative method instead (`method="peak"`). 

The detailed description of the outputs (returns) can be found in our documentation of the function [here](https://neurokit2.readthedocs.io/en/latest/functions.html#neurokit2.ecg.ecg_delineate)

hope that helps!

**Reply by sdujicun** [reply]
Thanks a lot.  Following the suggestion, my problem has been solved
